pkg/background/update_request_controller.go

```diff
@@ -232,10 +227,6 @@ func (c *Controller) updatePolicy(old, cur interface{}) {
 		}
 	}
 
-	if reflect.DeepEqual(curP.Spec, oldP.Spec) {
-		policyHasGenerate = false
-	}
```

Comment:
This update handler is only for generate policies. @prateekpandey14 - do we need it?

Reply:
i think we can remove it, the earlier we don't to enqueue the UR if there is no changes in Policy.Spec on update

Reply:
Policy has mutate rule, uses the UR to trigger mutate existing as well which is getting processed via different handler, i think we only wants to process the generate rule here.

Reply:
I removed the check.